doc/build/changelog/migration_20.rst

```diff
@@ -2095,6 +2095,38 @@ when using an explicit :func:`_orm.aliased` object, both from a user point
 of view as well as how the internals of the SQLAlchemy ORM must handle it.
 
 
+Filtering operations no longer accept explicit subqueries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Synopsis**
+
+Support for accepting explicit subqueries in filtering operations, such as
+`_sql.expression.ColumnOperators.in_` and
+`_sql.expression.ColumnOperators.not_in`, has been removed in 2.0 ::
+
+    subq = (
+        session.query(User.id)
+        .filter(User.name == "foo")
+    ).subquery()
+    q = (
+        session.query(User)
+        .filter(User.id.in_(subq))
+    )
+
+**Migration to 2.0**
+
+Under 2.0, SQLAlchemy will automatically interpret a fully constructed query
+as a subquery based on the implied context ::
+
+    subq = (
+        session.query(User.id)
```

Comment:
for migration to 2.0 the select() construct, not session.query(), should be used.   with select(), you'd want to call subquery() on that unless you are using it inside of in_().  so this note should also mention that this is specific to in_()/not_in()

Reply:
I'm going to move stuff around and incorporate your info.  I'll use github for a bit and try to avoid gerrit if you don't mind.

Reply:
I waited a bit before revisiting, and think this is the approach.  Ready for review.